lib/galaxy/datatypes/binary.py

```diff
@@ -203,14 +203,24 @@ class Bam( Binary ):
     MetadataElement( name="bam_header", default={}, desc="Dictionary of BAM Headers", param=MetadataParameter, readonly=True, visible=False, optional=True, no_value={} )
 
     def _get_samtools_version( self ):
-        # Determine the version of samtools being used.  Wouldn't it be nice if
-        # samtools provided a version flag to make this much simpler?
         version = '0.0.0'
         samtools_exec = which('samtools')
         if not samtools_exec:
             message = 'Attempting to use functionality requiring samtools, but it cannot be located on Galaxy\'s PATH.'
             raise Exception(message)
 
+        # Get the version of samtools via --version-only, if available
+        p = subprocess.Popen( ['samtools', '--version-only'],
```

Comment:
Well...so yes the first argument is that I am adapting the code to the context. I have seen that this function is using `subprocess.Popen()`, so I followed the same idea.

But the second argument is that I have spent quite a few time on the `subprocess` library lately, and I like the control it offers. In this case, I am storing the `error` if future modifications require it, but we could also bump it if you want.

And `check_output()` is a quite a specific use-case of `Popen`. In here, I would have needed to wrap the `check_output()` with a `try...catch` to handle the `CalledProcessError`...

So, I understand as we just need the output, it would make sense to call `check_output`, but I found it annoying to handle the exception and be "stuck" if suddenly we need `stderr`. (Plus it seems to be quicker to use `Popen`, as `check_output` is calling it behind the scene :D).

Thank for looking at the code!